File: php-poo/aulas/aula04-construct/Caneta.php
<?php

class Caneta
{
    private $modelo;
    private $cor;
    private $ponta;
    private $tampada;

//    // Método Construtor, 2º forma: colocando o nome da classe
//    public function Caneta() {
//        $this->cor = "Azul";
//        $this->tampar();
//    }

    // Método Construtor, 1º forma: __construct()
    public function __construct() {
        $this->cor = "Azul";
        $this->tampar();
    }

    public function tampar() {
        $this->tampada = true;
    }

    public function getModelo() {
        return $this->modelo;
    }

    public function setModelo($m) {
        $this->modelo = $m;
    }

    public function getPonta() {
        return $this->ponta;
    }

    public function setPonta($p) {
        $this->ponta = $p;
    }
}

File: java-poo/projects/poo-aula04/src/aula04/Caneta.java
package aula04;

public class Caneta {
    // Atributos
    private String modelo;
    private float ponta;
    private String cor;
    private boolean tampada;

    // Método Construtor
    public Caneta(String m, String c, float p) {
        this.modelo = m;
        this.cor = c;
        this.ponta = p; // this.setPonta(p);
        this.tampar();
    }


    // Métodos Getters e Setters
    public String getModelo() {
        return this.modelo;
    }
    public void setModelo(String m) {
        this.modelo = m;
    }
    public float getPonta() {
        return this.ponta;
    }
    public void setPonta(float p) {
        this.ponta = p;
    }

    // Método para tampar() e destampar() a caneta
    public void tampar() {
        this.tampada = true;
    }
    public void destampar() {
        this.tampada = false;
    }

    // Método para retornar o status da caneta
    public void status() {
        System.out.println("----------------------------------------");
        System.out.println("Modelo: " + this.getModelo());
        System.out.println("Ponta: " + this.getPonta());
        System.out.println("Cor: " + this.cor);
        System.out.println("Está tampada? " + this.tampada);
        // System.out.println("----------------------------------------");
    }
}
